src/distribution/chi_squared.rs

```diff
@@ -322,16 +326,20 @@ impl Continuous<f64, f64> for ChiSquared {
     /// ln(1 / (2^(k / 2) * Γ(k / 2)) * x^((k / 2) - 1) * e^(-x / 2))
     /// ```
     fn ln_pdf(&self, x: f64) -> f64 {
-        self.g.ln_pdf(x)
+        if x > 0.0 {
+            self.g.ln_pdf(x)
+        } else {
+            -f64::INFINITY
```

Comment:
conventionally we've been using `f64::NEG_INFINITY`,  I'm not actually 100% sure `-f64::INFINITY == f64::NEG_INFINITY`

Reply:
Sorry, I completely missed this one.

BTW, I didn't know that these constants are defined like this :grin: :

```rust
pub const INFINITY: f64 = 1.0f64 / 0.0f64
pub const NEG_INFINITY: f64 = -1.0f64 / 0.0f64
```